types/sindresorhus__df/index.d.ts

```diff
@@ -0,0 +1,23 @@
+// Type definitions for sindresorhus__df 2.10
+// Project: https://github.com/sindresorhus/df (Does not have to be to GitHub, but prefer linking to a source code repository rather than to a project website.)
+// Definitions by: Matt Thompson <https://github.com/whatknight>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.1
+export interface SpaceInfo {
+    filesystem: string;
+    size: number;
+    used: number;
+    available: number;
+    capacity: number;
+    mountpoint: string;
+}
+
+interface Df {
```

Comment:
Declaring an interface to represent the module shape is unnecessary, you can just export a function directly:
```ts
declare function df(): Promise<df.SpaceInfo[]>;
declare namespace df {
    function fs(filesystem: string): Promise<SpaceInfo>;
    function file(file: string): Promise<SpaceInfo>;
    interface SpaceInfo { ... }
}
```

Reply:
I'm new to TS. Since the function and namespace are named the same, when we export df they overload?

Reply:
They merge to become a single entity. https://www.typescriptlang.org/docs/handbook/declaration-merging.html